src/CMakeLists.txt

```diff
@@ -111,24 +113,36 @@ endif()
 
 if ("${RUNTIME_STATS}" MATCHES "ON")
   set(LEAN_EXTRA_CXX_FLAGS "${LEAN_EXTRA_CXX_FLAGS} -D LEAN_RUNTIME_STATS")
+  set(LEANC_EXTRA_C_FLAGS "${LEANC_EXTRA_C_FLAGS} -D LEAN_RUNTIME_STATS")
 endif()
 
 if (NOT("${CHECK_OLEAN_VERSION}" MATCHES "ON"))
   set(LEAN_EXTRA_CXX_FLAGS "${LEAN_EXTRA_CXX_FLAGS} -D LEAN_IGNORE_OLEAN_VERSION")
+  set(LEANC_EXTRA_C_FLAGS "${LEANC_EXTRA_C_FLAGS} -D LEAN_IGNORE_OLEAN_VERSION")
 endif()
 
-if("${CMAKE_C_COMPILER}" MATCHES "emcc")
-    set(EMSCRIPTEN ON)
+if(${CMAKE_SYSTEM_NAME} MATCHES "Emscripten")
     set(MULTI_THREAD OFF)
-    set(TCMALLOC OFF)
-    set(CFLAGS_EMSCRIPTEN "-Oz -O3")
-    set(LEAN_EXTRA_CXX_FLAGS "${LEAN_EXTRA_CXX_FLAGS} ${CFLAGS_EMSCRIPTEN} -D LEAN_EMSCRIPTEN")
-    set(LEAN_EXTRA_LINKER_FLAGS "${LEAN_EXTRA_LINKER_FLAGS} --memory-init-file 0 --llvm-lto 1 -s ALLOW_MEMORY_GROWTH=1 -s DISABLE_EXCEPTION_CATCHING=0 ${CFLAGS_EMSCRIPTEN}")
-    set(LEAN_JS_LIBRARY "${CMAKE_CURRENT_SOURCE_DIR}" CACHE STRING "location of olean files for lean.js")
+    # TODO(WN): code size/performance tradeoffs
+    # - we're using -O3; it's /okay/
+    # - -flto crashes at runtime
+    # - -Oz produces quite slow code
+    # - system libraries such as OpenGL are included in the JS but shouldn't be
```

Comment:
Wow, any idea why?

Reply:
So OpenGL in particular is included in the JS because I didn't get around to hand-picking the libraries we need. But I'm not worried about this as the vast majority of startup cost is due to WASM size, not the JS. Even when included initially, OpenGL would go away if we ran Emscripten's dead-code elimination, but we cannot because otherwise dynamic symbols which the interpreter needs (e.g. `Usize.ofNat32` extern) are also stripped. Leo mentioned you tried adding export-keepalive annotations on Windows but it didn't work? On Emscripten this is `EMSCRIPTEN_KEEPALIVE`.